### ISSUE TITLE
Integration of reactor with water fails below Temperature of 328 K

### ISSUE THREAD

**Issue by ulrich1a** [Cantera/cantera#475]
### Actual Behavior:
CanteraError thrown by Substance::set_T:
illegal temperature: 272.5

### Steps to reproduce: run the following python script
import cantera as ct
myWater = ct.Water()
myWater.TP = 327.0, 1.0*ct.one_atm 
r2 = ct.Reactor(contents=myWater)
sim = ct.ReactorNet([r2])
sim.advance(4.e-4)

**Comment by bryanwweber** [comment]
This is expected behavior. The minimum temperature allowed for the `Water` phase is 273.16 K, which you can check with

    >>> myWater.min_temp
    273.16

**Comment by bryanwweber** [comment]
On second thought, the `Reactor` shouldn't be doing anything. There aren't any energy transfers, so the temperature shouldn't decrease, so I'm not sure what's happening here. This might require looking into the solver. 

You can get this to advance if you turn off the energy equation:

    r2.energy_enabled = False
    sim = ct.ReactorNet([r2])
    sim.advance(4.0E-4)

Can you please give some more detail of your use case that raises this issue?

**Comment by ulrich1a** [comment]
I am interested in the temperature of a reactive gas pressurized in a reactor with water. The inflow of water is modeled with a valve. Also the heat transfer to the water is needed in this case. 
This worked for me with Cantera 2.2. So it looks like a regression.

**Comment by skyreflectedinmirrors** [comment]
@bryanwweber could this be due to the FD Jacobian?
If there's any non-zero (potentially spurious) derivative of the temperature, you would end up evaluating the RHS at a temperature ~ T + dT/dt * epsilon, which could account for the "drop"


edit: on second thought, that looks like a pretty large deltaT to be evaluating at

**Comment by bryanwweber** [comment]
@ulrich1a Can you provide a complete script that worked with Cantera 2.2?

**Comment by ulrich1a** [comment]
Here is a version, that worked with water, although I used here 
`myWater = ct.Solution('liquidvapor.xml', 'water')`
[gas_compression.zip](https://github.com/Cantera/cantera/files/1325090/gas_compression.zip)

**Comment by bryanwweber** [comment]
With the script you provided and Cantera 2.2.1 (and 2.2.0), I get an `updateState` error in the Reactor on macOS, so the script doesn't work for me. Can you confirm that it works for you?

**Comment by speth** [comment]
I also couldn't get this to work with Cantera 2.2, but the changes that generate the current behavior are the ones introduced in 1dc0cedcd93e8dd6b766b7109834ba18fc121646 which were implemented to resolve #257.

I think the issue here is that while the `bracket_and_solve_root` function is generally more robust than the Newton's method algorithm that was previously implemented, it does need some work so that it doesn't run into trouble trying to evaluate properties outside the valid temperature range for the state.

**Comment by ulrich1a** [comment]
The code in https://github.com/Cantera/cantera/commit/1dc0cedcd93e8dd6b766b7109834ba18fc121646 contains the line: 

```diff
+            TT = bmt::bracket_and_solve_root(
+                u_err, T, 1.2, true, bmt::eps_tolerance<double>(48), maxiter);
```
The reported temperature in the error message is just by a factor 1/1.2 smaller than the starting temperature. So for me it seems, that the factor 1.2 hardcoded in the function call has to be adjusted in case of pure liquids to avoid a temperature outside of the limits.

The following script was running without error in version 2.2.1:
```python
# -*- coding: utf-8 -*-

import cantera as ct

#myWater = ct.Water()
myWater = ct.Solution('liquidvapor.xml', 'water')
#myWater.TP = 328.0, 7.0*ct.one_atm 
#myWater.TX = 328.0, 1.0 
myWater.TPX = (300.0, 7.0*ct.one_atm, 'H2O:1.0') # Temperatur 35°C
r2 = ct.Reactor(contents=myWater)
#r2.volume = 0.000001
r2.syncState()

print('Temperatur 1:',r2.T)
print(r2.thermo.P)
sim = ct.ReactorNet([r2])
sim.verbose = True
#print(r2.get_state())
sim.advance(4.e-4)
print('Temperatur 2:',r2.T)
print(r2.thermo.P)
```
